bin/wasm-node/rust/src/json_rpc_service.rs

```diff
@@ -231,22 +264,22 @@ struct JsonRpcService {
     /// For each active finalized blocks subscription (the key), a sender. If the user
     /// unsubscribes, send the unsubscription request ID of the channel in order to close the
     /// subscription.
-    all_heads: Mutex<HashMap<String, oneshot::Sender<String>>>,
+    all_heads: Mutex<HashMap<u32, oneshot::Sender<String>>>,
```

Comment:
```suggestion
    all_heads: Mutex<HashMap<SubscriptionId, oneshot::Sender<String>>>,
```

I don't understand why you replaced some `String` with `u32` and some others with `SubscriptionId`. They should all be `SubscriptionId`.